var/spack/repos/builtin/packages/r-biomart/package.py

```diff
@@ -42,13 +42,15 @@ class RBiomart(RPackage):
     homepage = "https://bioconductor.org/packages/biomaRt/"
     git      = "https://git.bioconductor.org/packages/biomaRt.git"
 
+    version('2.36.1', commit='5634e57e20199f9dc1f8b927eb3893143fc02f4f')
     version('2.34.2', commit='a7030915fbc6120cc6812aefdedba423a207459b')
     version('2.32.1', commit='f84d74424fa599f6d08f8db4612ca09914a9087f')
 
     depends_on('r-xml', type=('build', 'run'))
     depends_on('r-rcurl', type=('build', 'run'))
     depends_on('r-annotationdbi', type=('build', 'run'))
-    depends_on('r-progress', type=('build', 'run'), when='@2.34.2')
-    depends_on('r-stringr', type=('build', 'run'), when='@2.34.2')
-    depends_on('r-httr', type=('build', 'run'), when='@2.34.2')
-    depends_on('r@3.4.0:3.4.9', when='@2.32.1:')
+    depends_on('r-progress', type=('build', 'run'), when='@2.34.2:')
+    depends_on('r-stringr', type=('build', 'run'), when='@2.34.2:')
+    depends_on('r-httr', type=('build', 'run'), when='@2.34.2:')
+    depends_on('r@3.4.0:3.4.9', when='@2.32.1:2.35', type=('build', 'run'))
+    depends_on('r@3.5.0:3.5.9', when='@2.36.1', type=('build', 'run'))
```

Comment:
just use an open range here

Reply:
Sorry, I'm not following what you're asking for. Are you saying leave the 3.5 dependency unbound on the high end ?

For reference:
```
biomart 2.32.1 = bioconductor 3.5 = r 3.4
biomart 2.34.2 = bioconductor 3.6 = r 3.4
biomart 2.36.1 = bioconductor 3.7 = r 3.5
```

Reply:
In general when introducing a new version with changed dependencies I would 
1) close the now no longer correct dependency ranges (in this case adding the `2.35` to the `@2.32.1:2.35`), preferably with the exact version where this dependency changed, but as @adamjstewart said: This might be tricky to track down exactly
2) add the new depends_on entry, assuming that the package doesn't change it dependencies for future versions, i.e. adding all new dependencies as upwards open ranges

Unless you know that this exact package version can only exactly work with another package version, using exact versions will be painful to maintain. And note that `@2.35` does not match `version('2.35.0')`, what I would (naively [1]) expect is that `biomart@2.32.X` is only compatible with `bioconductor@3.5.Y` for all X and Y. In that case you should write the dependency as `depends_on('bioconductor@3.5.0:3.5.99', when='@2.32.0:2.32.99')`.


[1] without knowing anything about biomart or bioconductor (or r), but that is how versions are typically used, i.e. bugfixes don't affect inter usability. Technically minor versions should be strictly forward compatible (i.e. you may depend on a minimal version because you need a new feature from it, but a newer minor version may not remove any interface), but ... reality and such

Reply:
thanks for the note about 2.35 <> 2.35.0, didn't realize that, I'll update.

bioconductor versions are terrible...

Reply:
also, to be clear - there is no `bioconductor 3.5.Y`, just 3.5, 3.6, etc. where did `biomart 2.33.X` go? did it ever exist, and if so was it part of bioconductor 3.5 or 3.6 ? who knows without digging through their commit history